catalog/harvest/catalog-harvest-common/src/main/java/org/codice/ddf/catalog/harvest/common/FileSystemPersistenceProvider.java

```diff
@@ -93,16 +96,22 @@ String getMapStorePath() {
   @Override
   public void store(String key, Object value) {
     File dir = new File(getMapStorePath());
-    if (!dir.exists() && !dir.mkdir()) {
-      LOGGER.debug("Unable to create directory: {}", dir.getAbsolutePath());
-    }
-    LOGGER.trace("Entering: store - key: {}", key);
-    try (OutputStream file = new FileOutputStream(getMapStorePath() + key + PERSISTED_FILE_SUFFIX);
-        OutputStream buffer = new BufferedOutputStream(file);
-        ObjectOutputStream output = new ObjectOutputStream(buffer)) {
-      output.writeObject(value);
-    } catch (IOException e) {
-      LOGGER.debug("IOException storing value in cache with key = " + key, e);
+
+    if (new File(getPersistencePath()).canWrite()) {
+      boolean created = dir.mkdirs();
+      if (created || dir.exists()) {
+        LOGGER.trace("Entering: store - key: {}", key);
+        try (OutputStream file =
+                new FileOutputStream(getMapStorePath() + key + PERSISTED_FILE_SUFFIX);
+            OutputStream buffer = new BufferedOutputStream(file);
+            ObjectOutputStream output = new ObjectOutputStream(buffer)) {
+          output.writeObject(value);
+        } catch (IOException e) {
+          LOGGER.debug("IOException storing value in cache with key = " + key, e);
```

Comment:
❗️Failing to write to the file system is pretty bad and there may be something an admin can do about it so we should consider logging something higher than `debug` here and on L113.

Reply:
✏️ `LOGGER.debug("IOException storing value in cache with key = {}", key, e);`